codegenerator/cli/templates/static/codegen/src/eventFetching/ChainManager.res

```diff
@@ -72,12 +72,35 @@ type fetchStateWithData = {
   heighestBlockBelowThreshold: int,
 }
 
+let isQueueItemEarlierUnorderedBelowReorgThreshold = (
+  ~fetchStatesMap: ChainMap.t<fetchStateWithData>,
+) => (a: multiChainEventComparitor, b: multiChainEventComparitor) => {
+  let isItemBelowReorgThreshold = item => {
+    item.earliestEvent->FetchState.queueItemIsInReorgThreshold(
+      ~heighestBlockBelowThreshold=ChainMap.get(
+        fetchStatesMap,
+        a.chain,
+      ).heighestBlockBelowThreshold,
+    )
+  }
+  switch (a->isItemBelowReorgThreshold, b->isItemBelowReorgThreshold) {
+  | (false, true) => true
+  | (true, false) => false
+  | _ => isQueueItemEarlierUnorderedMultichain(a, b)
```

Comment:
Cool, it makes sense that if either both chains are below the threshold or both chains are above the threshold, they are treated equally.
 
(Two quick thoughts that aren't easy to validate from a GitHub PR.)
1.  Once the reorg threshold is met (on all chains), and it starts using the entity history table - does it stop using this function? I guess this question is asing if https://github.com/enviodev/hyperindex/pull/268/files#r1803437498 propogates through to this function.

2.  What about the case where one chain is intermittently faulty? All of the chains sync to the head (including the faulty one) and then that faulty one chain starts falling behind. Will it cause non-faulty chains to also only process up until the reorg threshold once again? Or it stops using this function, and that isn't an issue (I guess back to point 1).

Reply:
1. Yes it will stop using this function due to this check here: https://github.com/enviodev/hyperindex/blob/dfe22696676d648ce5d4748f4c4d6ed7ce2df5e2/codegenerator/cli/templates/static/codegen/src/eventFetching/ChainManager.res#L102-L110

Once we hit the reorg threshold for the first time it will start using the other functions.

2. During the historical sync, the other chains will keep processing below their reorg threshold until the faulty chain reaches a block that's comparable with the others. This is only during historical sync, then all chains can be processed in any order. Lmk know if you can think of a better way to handle this but I tried a few options and this was the best I came up with.